### ISSUE TITLE
Issue upgrading from 3.14.1 to 3.14.2

### ISSUE THREAD

**Issue by frederikhors** [rob-balfre/svelte-select#213]
Reproduction: https://svelte.dev/repl/800b7f24191848ef8c7984c1ff9edf1e?version=3.31.0

Using this:

```js
  const getOptionLabel = (option) => {
    return option.username
  }
```

I get `undefined` now instead of the string.

I'm using this object:

```js
{id: 123, username: "John"}
```

When I open my svelte page with Svelte-Select component in console.log I see:

```log
selectedValue: {id: 123, username: "John"}
selectedValue: {id: "", label: ""}
```

Why?

With `svelte-select@3.14.1` it doesn't happen.

![image](https://user-images.githubusercontent.com/41120635/102000305-7ea7c180-3ce6-11eb-82e7-ba02e484a486.png)

**Comment by frederikhors** [comment]
> When I open my svelte page with Svelte-Select component in console.log I see:
> 
> ```
> selectedValue: {id: 123, username: "John"}
> selectedValue: {id: "", label: ""}
> ```

Why @stephenlrandall?

**Comment by frederikhors** [comment]
@stephenlrandall NOW I GOT IT! https://svelte.dev/repl/800b7f24191848ef8c7984c1ff9edf1e?version=3.31.0

**Comment by stephenlrandall** [comment]
I think I understand what's happening. The 

> @stephenlrandall NOW I GOT IT! https://svelte.dev/repl/800b7f24191848ef8c7984c1ff9edf1e?version=3.31.0

Your `optionIdentifier` is set to `id` so when you set `selectedValue` to an object with an `id` value not in `items`, what is the expected behavior?

**Comment by frederikhors** [comment]
I'm using it like this because when my I open the page the items are reloading in background; after the reload the user is still there. I don't understand what is happening.

**Comment by frederikhors** [comment]
Your change should consider this usecase.

**Comment by frederikhors** [comment]
Are you resetting it to `{label: "", id: ""}`?

**Comment by frederikhors** [comment]
It think this behavior should not be the default.

**Comment by frederikhors** [comment]
This is not the problem I think. I think something different is going on, the REPL is not 100% what is on my local. Gimme more time, please.

**Comment by frederikhors** [comment]
You're right. The problem is on my side I think.

When I open form page the `<Select/>` fetch a few users which have ID different from the one selected.

With the previous 3.14.1 the problem was not visible I think because it wasn't updating the selectedValue and as soon as I clicked the query was getting my user.

Or at least I think it is like this. I'm sad. :(

**Comment by frederikhors** [comment]
But I still think this is not a svelte-select thing to do. Why it changes my `selectedValue`? It cannot do that until I click on it. Am I wrong, @rob-balfre, @stephenlrandall?

3.14.1 was like this.

Now what I have to do?

If I don't want to change it when select appear on the page?

**Comment by frederikhors** [comment]
It is updating `selectedValue` a lot of times! Even in REPL at least 3 times. Why?

**Comment by rob-balfre** [comment]
It's the two way bind, I'd suggest using `on:select` if it's causing you issues. With Svelte any child components (not just svelte-select) will fire reactive events once for each `bind:` - furthest child first then moving up until it hits the root component.

Also there is hidden logic inside svelte-select to cater for all the different modes and props, so the value can mutate and confuse stores if two way binding it used.

**Comment by rob-balfre** [comment]
@frederikhors for example... https://svelte.dev/repl/5f64f34f2aa7499ca261c9a1c3ba0ad6?version=3.31.0

**Comment by frederikhors** [comment]
Ok. Good on:select, I can try it. What scares me is why it changes my selectedValue. Can we do something for this?

**Comment by frederikhors** [comment]
With 3.14.1 it was fine.

**Comment by rob-balfre** [comment]
A patch release with breaking changes ... what you complaining about 😂 

Yeah I guess that does need a fix... hmm

**Comment by frederikhors** [comment]
> Yeah I guess that does need a fix

And a good example on how to handle this case. Because I can see the value, but I don't understand where I'm wrong.

**Comment by rob-balfre** [comment]
@frederikhors should be fixed in v3.14.3?

**Comment by frederikhors** [comment]
It works now! Thanks a lot!
